docs/blog/_posts/2019-12-20-21th-dotty-milestone-release.md

```diff
@@ -0,0 +1,378 @@
+  case "empty" => 0
+  case "nonempty" => 1
+}
+```
+
+By using the new treatment we can now offer `match` as a method:
+
+```scala
+xs.match {
+  case Nil => false
+  case _ => true
+}
+```
+
+You can read more on our docs [Match Expressions](https://dotty.epfl.ch/docs/reference/changed-features/match-syntax.html) and on the interesting discussions on [contributors](https://contributors.scala-lang.org/t/pre-sip-demote-match-keyword-to-a-method/2137/2).
+
+## New quoted pattern matching
+
+We introduce a high-level API to deconstruct or extract values out of `Expr`
+using pattern matching. It consists of high-level extractors for getting static
+information out of exprs and, of quoted patterns that allows to deconstruct
+complex code that contains a precise structure, types or methods.
+Patterns `'{ ... }` can be placed in any location where Scala expects a pattern.
+
+The new extractors are summarized below:
+
+* `scala.quoted.matching.Const`: matches an expression a literal value and returns the value.
+* `scala.quoted.matching.ExprSeq`: matches an explicit sequence of expresions and returns them. These sequences are useful to get individual `Expr[T]` out of a varargs expression of type `Expr[Seq[T]]`.
+* `scala.quoted.matching.ConstSeq`:  matches an explicit sequence of literal values and returns them.
+
+The following snippet demonstrates the new _quoted patterns_ implementing a
+simple, 1-level, non-recursive rewriter macro for exponents. `rewrite` is a an
+inline method definition designating a macro as usual. To inspect an `expr`
+value with friendly syntax we can now use the quoted syntax as patterns inside a
+match expression. Notice that quotes designate patterns and `$`, the familiar
+syntax for splices is used to _extract_ (capture) information out of a pattern.
+
+```scala
+inline def rewrite(expr: => Double): Double = ${rewrite('expr)}
+
+def rewrite(expr: Expr[Double])(given QuoteContext): Expr[Double] = {
+  val res = expr match {
+    // product rule
+    case '{ power2($a, $x) * power2($b, $y)} if a.matches(b) => '{ power2($a, $x + $y) }
+    // rules of 1
+    case '{ power2($a, 1)} => a
+    case '{ power2(1, $a)} => '{ 1.0 }
+    // rule of 0
+    case '{ power2($a, 0)} => '{ 1.0 }
+    // power rule
+    case '{ power2(power2($a, $x), $y)} => '{ power2($a, $x * $y ) }
+    case _ => expr
+  }
+  res
+}
+```
+
+To learn more read our docs on [pattern matching over quotes](https://dotty.epfl.ch/docs/reference/metaprogramming/macros.html#pattern-matching-on-quoted-expressions).
+
+## Added support for SemanticDB extraction
+
+Scala 3 now offers support for SemanticDB extraction producing `.semanticdb`
+files. Users can use the `-Ysemanticdb` compiler flag to enable exporting
+SemanticDB files. Providing `-semanticdb-target` allows the user to select a
+separate target destination for the `META-INF` directory that stores SemanticDB
+files and `-sourceroot` to calculate a relative path for SemanticDB files within
+`META-INF`.
+
```

Comment:
as I also suggested above, the relation to Metals should be made explicit

Reply:
thanks